ecr-login/api/factory_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package api
+
+import (
+	"os"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSharedConfigStateDefaultBehavior(t *testing.T) {
+	os.Unsetenv("AWS_SDK_LOAD_CONFIG") // Ensure the environment variable is not set
```

Comment:
For each of these, I would recommend reading and restoring the original environment variable state.

You could also convert this into a single table-driven test (with Go subtests) since they have the same setup logic and assertions.

```go
const loadConfigEnvVar = "AWS_SDK_LOAD_CONFIG"

func TestSharedConfigState(t *testing.T) {
	defer func(value string) {
		os.Setenv(loadConfigEnvVar)
	}(os.Getenv(loadConfigEnvVar)

	cases := []struct{
		envValue string
		expected session.SharedConfigState
	}{
		{"", session.SharedConfigEnable},
		{"true", session.SharedConfigEnable},
		{"false", session.SharedConfigEnable},
	}

	for _, testCase := range cases {
		t.Run(testCase.envValue, func(t *testing.T) {
			os.Setenv(loadConfigEnvVar, testCase.envValue)
			state := loadSharedConfigState()
			assert.NotNil(t, state)
			assert.Equal(t, testCase.expected, state)
		})
	}
}